src/main/java/io/jenkins/pluginhealth/scoring/probes/DeprecatedPluginProbe.java

```diff
@@ -0,0 +1,55 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(value = DeprecatedPluginProbe.ORDER)
+public class DeprecatedPluginProbe extends Probe implements UpdateCenterProbe {
+    public static final int ORDER = CompoundUpdateCenterProbes.ORDER + 1;
+
+    public static final String DEPRECATION_KEY = "deprecation";
+
+    private final ObjectMapper objectMapper;
+    private final String updateCenterURL;
+    private UpdateCenter updateCenter;
+
+    public DeprecatedPluginProbe(@Value("${jenkins.update-center}") String updateCenterURL) {
+        this.objectMapper = Jackson2ObjectMapperBuilder.json().build();
+        this.updateCenterURL = updateCenterURL;
+    }
+
+    record UpdateCenterDeprecations(String url) {
+    }
+
+    record UpdateCenter(Map<String, UpdateCenterDeprecations> deprecations) {
+    }
+
+    @Override
+    public void readUpdateCenter(Plugin plugin) throws IOException {
+        updateCenter = objectMapper.readValue(new URL(updateCenterURL), UpdateCenter.class);
+    }
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin) {
+        if (updateCenter.deprecations().containsKey(plugin.getName())) {
+            return ProbeResult.failure(key(), updateCenter.deprecations().get(plugin.getName()).url());
+        }
+        return ProbeResult.success(key(), "This plugin is NOT deprecated");
+    }
```

Comment:
Is this correct?
We are not reading the `update-center.json` file inside the `doApply` method, instead, we're using the `updateCenter` object which contains the representation which was previously read.

Reply:
the idea is to only download and parse the `update-center.json` once for all the probes that requires it.

Reply:
How would I parse the `update-center.json` just once at the beginning for all the probes that require it?

The `readValue` method's second argument is taking the Class which it will use as a blueprint to extract data from the JSON object.

And since the data-to-be-extracted varies from probe to probe (probe related to deprecations doesn't need "plugins" object, probe related to adoption needs the object "plugins"->"labels"), I'm confused about which class should I pass as the 2nd arg to the `readValue` method? Currently, we're using `records` as an ad-hoc replacement for POJO, but I don't think that would be the way to go in this scenario.

Reply:
I'm working on this PR to match this and even improve #32.